xsd-core/src/main/java/io/ballerina/xsd/core/XSDToRecord.java

```diff
@@ -197,6 +224,19 @@ private static Response generateTypes(XSDVisitor xsdVisitor,
         return new Response(generatedTypes, diagnostics);
     }
 
+    private static void handleExistingTypes(Map<String, ModuleMemberDeclarationNode> typesMap,
+                                            XSDVisitor xsdVisitor, ArrayList<String> existingTypes,
+                                            Map<String, ModuleMemberDeclarationNode> nodes) {
```

Comment:
I think we can relax the type of the existingTypes to be a collection as well
```suggestion
    private static void handleExistingTypes(Map<String, ModuleMemberDeclarationNode> typesMap,
                                            XSDVisitor xsdVisitor, Collection<String> existingTypes,
                                            Map<String, ModuleMemberDeclarationNode> nodes) {
```